ncs/Kconfig

```diff
@@ -8,10 +8,14 @@
 config SSF_SUIT_SERVICE_ENABLED
     bool
 
+config SUIT_RECOVERY
+  bool "The image is part of the SUIT recovery firmware"
+
 config SUIT_ENVELOPE_TEMPLATE
   string "Path to the envelope template"
   default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/app_envelope.yaml.jinja2" if SOC_NRF54H20_CPUAPP
-  default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/rad_envelope.yaml.jinja2" if SOC_NRF54H20_CPURAD
+  default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/rad_envelope.yaml.jinja2" if SOC_NRF54H20_CPURAD && !SUIT_RECOVERY
+  default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/rad_recovery_envelope.yaml.jinja2" if SOC_NRF54H20_CPURAD && SUIT_RECOVERY
```

Comment:
Maybe it would be better to reorder it as such:
```
  default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/app_envelope.yaml.jinja2" if SOC_NRF54H20_CPUAPP
  default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/rad_recovery_envelope.yaml.jinja2" if SOC_NRF54H20_CPURAD && SUIT_RECOVERY
  default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/rad_envelope.yaml.jinja2" if SOC_NRF54H20_CPURAD
```
That way the `!SUIT_RECOVERY` becomes unnecessary.